_posts/2020-04-17-snowplow-amp-tracker-1-0-1-released/index.md

```diff
@@ -0,0 +1,186 @@
+
+Note that while the AMP tracker attempts to retain the domain userid across pages and sessions, the AMP platform does not offer any means to guarantee that this can be done in all cases - so the user identification strategy here is based on having the value attached to at least one event, rather than all events.
+
+<h3 id="amp-linker"> 2.2 AMP Linker</h3>
+
+The AMP tracker now offers the ability to attach the AMP client ID to the querystring, in order to identify users moving from an AMP page to a non-AMP page. This is enabled by ensuring that the AMP linker is enabled for any destination domains required:
+
+{% highlight html %}
+...
+"linkers": {
+  "enabled": true,
+  "proxyOnly": false,
+  "destinationDomains": ["ampdomain"]
+},
+...
+{% endhighlight %}
+
+This will add a querystring parameter ‘linker=’ to the destination url, which contains the amp_id value, base-64 encoded.
+
+
+<h3 id="modeling-users"> 2.3 Modeling Considerations</h3>
+
+The value attached to the querystring by the AMP linker will look something like this: `?linker=1*1c1wx43*amp_id*amp-a1b23cDEfGhIjkl4mnoPqr`. To extract the AMP ID, this must be parsed, and the value immediately following the `amp_id*` string must be extracted.
+
+This value will only be present for the first page the user lands on after leaving the AMP page.
+
+If a `domain_userid` is found by the AMP tracker, it is not guaranteed to be attached to every event. Therefore, a good strategy for modeling user identification on both sides is to create a mapping table of domain_userid to amp-id, and join this to the rest of the data to attribute users.
+
+
+<h2 id="custom-events">3. Custom Events</h2>
+
+Custom events and entities can now be tracked using the AMP tracker.
+
+Custom events are sent by instrumenting the selfDescribingEvent request in a trigger, and passing customEventSchemaVendor, customEventSchemaName, customEventSchemaVersion, customEventSchemaData to it as variables - where customEventSchemaData is an escaped JSON string, as follows:
+
+{% highlight html %}
+<amp-analytics type="snowplow_v2" id="custEvent">
+<script type="application/json">
+{
+  "vars": {
+    "collectorHost": "snowplow-collector.acme.com",  // Replace with your collector host
+    "appId": "campaign-microsite"                    // Replace with your app ID
+  },
+  "triggers": {
+    "trackSelfDescribingEvent": {
+      "on": "click",
+      "selector": "a",
+      "request": "selfDescribingEvent",
+      "vars": {
+        "customEventSchemaVendor": "com.snowplowanalytics.snowplow",
+        "customEventSchemaName": "link_click",
+        "customEventSchemaVersion": "1-0-1",
+        "customEventSchemaData": "{\"targetUrl\":\"${targetUrl}\",\"elementId\":\"TEST\",\"elementContent\":\"${elementContent}\"}"
+      }
+    }
+  }
+}
+</script>
+</amp-analytics>
+{% endhighlight %}
+
+Documentation can be found [here][cust-events-docs]
```

Comment:
Another use of "here" as a link (see comment above).  No big deal, just pointing it out.